src/lighting.h

```diff
@@ -0,0 +1,27 @@
+/*
+Minetest
+Copyright (C) 2021 x2048, Dmitry Kostenko <codeforsmile@gmail.com>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#pragma once
+
+/** Describes ambient light settings for a player
+ */
+struct Lighting
+{
+    float shadow_intensity {0.0f};
```

Comment:
I prefer an opt-out. With this shadows are off, until a game specifically opts in.

By default many games will (presumably) not care.
That's why I liked the multiplier before, there was an easy way expressing that a games does not care.

As is, most folks will never see shadows.

Reply:
We could set 0.3 here as a default that can be corrected via gamma. The thing I like about gamma is that it does not change either of 0.0 and 1.0 values, unlike a multiplier.

Reply:
Another option is to add a few lines of code to the minetest default game and set the shadow strength based on cloud density. Something perhaps as simple as this:
[shadow.txt](https://github.com/minetest/minetest/files/7877864/shadow.txt)